config/global.ini.php

```diff
@@ -877,6 +877,13 @@
 ; so it can be disabled here if necessary.
 disable_tracking_matomo_app_links = 0
 
+; Add a query hint for the order of joined tables when building segment queries in MySQL. This can be used to override
+; sub-optimal choices by the MYSQL optimizer and always ensure the query plan starts with the first table in the query.
+enable_segment_first_table_join_prefix = 0
+
+; Add a query hint for the order of the first table for all log table queries in MySQL.
+enable_first_table_join_prefix = 0
+
```

Comment:
Was wondering if this would make more sense to be part of the DB or DbReader config. As it's more related to database than general config. But don't have a strong opinion here

Reply:
Yes, that probably makes the most sense :+1:
I've moved the settings to the [database] section, as part of this I've made the `GeneralConfig` class a bit easier to extend and added a new descendant class `DatabaseConfig` so we can easily get settings from the `[database]` section in the same way as we already do for `[General]`